tests/test_mariadb.py

```diff
@@ -317,9 +317,15 @@ def test_mariadb_upgrade(
     mounts: List[Union[BindMount, ContainerVolume]] = [
         BindMount(host_path=tmp_path, container_path="/var/lib/mysql")
     ]
+    mariadb_old_images = {
+        "15.7": "registry.suse.com/suse/mariadb:11.4",
+        "16.0": "registry.suse.com/suse/mariadb:11.4",
+    }
+    # gosu was renamed to idexec from mariadb_version 10.11
+    gosu = "gosu" if OS_VERSION == "15.6" else "idexec"
     mariadb_old = DerivedContainer(
-        base="registry.suse.com/suse/mariadb:10.6",
-        containerfile='RUN set -euo pipefail; head -n -1 /usr/local/bin/gosu > /tmp/gosu;  echo \'exec setpriv --pdeathsig=keep --reuid="$u" --regid="$u" --clear-groups -- "$@"\' >> /tmp/gosu;  mv /tmp/gosu /usr/local/bin/gosu; chmod +x /usr/local/bin/gosu',
+        base=mariadb_old_images.get(OS_VERSION),
```

Comment:
it is always the same start image now - just hardcode it here rather than having that dict-lookup indirection?